### PR TITLE
Js linting

### PR DESCRIPTION
**Pull request by cdmo**
Let's figure out what to do with `233-jest` first

### CODE REVIEW

**Comment by cdmo** [comment]
```
➜  psulib_blacklight git:(js-linting) yarn lint
...
✖ 118 problems (118 errors, 0 warnings)
```
😰

**Comment by cdmo** [comment]
Most of the problems were either undefined vars that are globals (jquery and blacklight) and max length violations.

Playing around with this some, there is an additional "cyclomatic complexity" cop (err, I mean rule). See https://eslint.org/docs/rules/complexity. When I turn this on and set the level to 2 we do encounter several violations. I kinda like this rule.

Also, we can run a "Fix" with this tool too, though I haven't gotten it to do much:

`node_modules/.bin/eslint --fix --ext .js app/javascript spec/javascript`

Probably because the errors we had couldn't be fixed automatically.

FWIW: there is no "generate todo" kinda feature, see https://github.com/eslint/eslint/issues/9844

File: app/javascript/psulib_blacklight/javascripts/availability.js [hunk 1]
```diff
@@ -349,7 +361,7 @@ function availabilityDataStructurer(holdingMetadata) {
 
             // Supplement data with an itemType and remove callNumber conditionally
             holdingMetadata[libraryID].forEach(function(element) {
-                element.itemType = (element.itemTypeID in allItemTypes) ? allItemTypes[element.itemTypeID] : "";
+                element.itemType = (element.itemTypeID in allItemTypes) ? element.itemTypeID : "";
```
Comment (banukutlu) [inline app/javascript/psulib_blacklight/javascripts/availability.js]: @cdmo shouldn't this still be `allItemTypes[element.itemTypeID]`?

Comment (cdmo) [reply app/javascript/psulib_blacklight/javascripts/availability.js]: Good catch, yes, sorry for the mistake

Comment (banukutlu) [reply app/javascript/psulib_blacklight/javascripts/availability.js]:
```suggestion
                element.itemType = (element.itemTypeID in allItemTypes) ? allItemTypes[element.itemTypeID] : "";
```
per our conversation with @cdmo, reverting this back

**Pull request merged**